File: 0931_Minimum Falling Path Sum/MinimumFallingPathSum_NoCopy.cs
public class Solution
{
    public int MinFallingPathSum(int[][] matrix)
    {
        for (int i = 1; i < matrix.Length; i++)
            for (int j = 0; j < matrix[i].Length; j++)
                matrix[i][j] += Math.Min(matrix[i - 1][j], Math.Min(matrix[i - 1][ Math.Max(0, j - 1)], matrix[i - 1][Math.Min(j + 1, matrix[i].Length - 1)]));

        return matrix[matrix.Length - 1].Min();
    }
}

File: 0931_Minimum Falling Path Sum/MinimumFallingPathSum_NoCopy.java
class Solution {
    public int minFallingPathSum(int[][] matrix) {
        int m = matrix.length, n = matrix[0].length;
        for(int row=1;row<m;row++) {
            for(int col=0;col<n;col++) {
                   int leftTop = col == 0 ? Integer.MAX_VALUE : matrix[row-1][col-1];
                   int top = matrix[row-1][col];
                   int rightTop = col == n-1 ? Integer.MAX_VALUE : matrix[row-1][col+1];
                   matrix[row][col] += ( Math.min(leftTop, Math.min(top, rightTop)));
            }
        }

        int ans = Integer.MAX_VALUE;
        for(int i=0;i<n;i++) {
            ans = Math.min(ans, matrix[m-1][i]);
        }
        return ans;
    }
}